packages/stream_chat/lib/src/api/channel.dart

```diff
@@ -1643,47 +1643,47 @@ class ChannelClientState {
   /// Update channelState with updated information
   void updateChannelState(ChannelState updatedState) {
     final newMessages = <Message>[
-      ...updatedState?.messages ?? [],
+      ...updatedState.messages,
       ..._channelState?.messages
-              ?.where((m) =>
+              .where((m) =>
                   updatedState.messages
-                      ?.any((newMessage) => newMessage.id == m.id) !=
+                      .any((newMessage) => newMessage.id == m.id) !=
                   true)
-              ?.toList() ??
+              .toList() ??
           [],
-    ]..sort(_sortByCreatedAt);
+    ]..sort(_sortByCreatedAt as int Function(Message, Message)?);
```

Comment:
I think the function param should be made typed rather than castingit here.
```dart
int _sortByCreatedAt(Message a, Message b) =>
      a.createdAt.compareTo(b.createdAt);
```